sqlobject/src/main/java/org/jdbi/v3/sqlobject/internal/SqlObjectInitData.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.sqlobject.internal;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+import java.util.function.UnaryOperator;
+
+import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.extension.ExtensionMethod;
+import org.jdbi.v3.core.extension.HandleSupplier;
+import org.jdbi.v3.core.internal.exceptions.Sneaky;
+import org.jdbi.v3.sqlobject.GenerateSqlObject;
+import org.jdbi.v3.sqlobject.Handler;
+
+public final class SqlObjectInitData {
+    private static final Object[] NO_ARGS = new Object[0];
+    public static final ThreadLocal<SqlObjectInitData> INIT_DATA = new ThreadLocal<>();
+
+    public final boolean concrete;
+    public final Class<?> extensionType;
+    public final UnaryOperator<ConfigRegistry> instanceConfigurer;
+    public final Map<Method, UnaryOperator<ConfigRegistry>> methodConfigurers;
+    public final Map<Method, Handler> methodHandlers;
+
+    public SqlObjectInitData(
+            Class<?> extensionType,
+            UnaryOperator<ConfigRegistry> instanceConfigurer,
+            Map<Method, UnaryOperator<ConfigRegistry>> methodConfigurers,
+            Map<Method, Handler> methodHandlers) {
+        concrete = isConcrete(extensionType);
+        this.extensionType = extensionType;
+        this.instanceConfigurer = instanceConfigurer;
+        this.methodConfigurers = methodConfigurers;
+        this.methodHandlers = methodHandlers;
+    }
+
+    public static boolean isConcrete(Class<?> extensionType) {
+        return extensionType.getAnnotation(GenerateSqlObject.class) != null;
+    }
+
+    public static SqlObjectInitData initData() {
+        final SqlObjectInitData result = INIT_DATA.get();
+        if (result == null) {
+            throw new IllegalStateException("Implemented SqlObject types must be initialized by SqlObjectFactory");
+        }
+        return result;
+    }
+
+    public static Method lookupMethod(String methodName, Class<?>... parameterTypes) {
+        return lookup0(initData().extensionType, methodName, parameterTypes);
+    }
+
+    private static Method lookup0(Class<?> klass, String methodName, Class<?>... parameterTypes) {
```

Comment:
not a fan of the `0` suffix for internal delegation, there must be a more meaningful name you can give such methods

Reply:
That's how the JDK does it... I can use `unsafe` instead I guess.